quaddtype/numpy_quaddtype/src/casts.cpp

```diff
@@ -605,6 +605,175 @@ quad_to_bytes_loop(PyArrayMethod_Context *context, char *const data[],
     return 0;
 }
 
+// StringDType to QuadDType casting
+static NPY_CASTING
+stringdtype_to_quad_resolve_descriptors(PyObject *NPY_UNUSED(self), PyArray_DTypeMeta *dtypes[2],
+                                        PyArray_Descr *given_descrs[2], PyArray_Descr *loop_descrs[2],
+                                        npy_intp *view_offset)
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = given_descrs[0];
```

Comment:
If you do this after checking `given_descrs[1]` then there's no need to decref in the error paths below so it'll be a little clearer